tdrs-backend/tdpservice/search_indexes/templates/multiselectdropdownfilter.html

```diff
@@ -1,48 +1,52 @@
 {% load i18n admin_urls %}
-<h3>{% blocktrans with filter_title=title %} By {{ filter_title }} {% endblocktrans %}</h3>
+<div class="list-filter-dropdown">
+    <h3>{% blocktrans with filter_title=title %} By {{ filter_title }} {% endblocktrans %}</h3>
 
-<div>
-    {% for choice in choices|slice:":1" %}
-        <a href="{{ choice.query_string|iriencode }}" title="Show {{ choice.display }}">Show {{ choice.display }}</a>
-    {% endfor %}
-    <select id="{{ title|slugify }}_select" multiple style="color:#000000;width:100%" aria-multiselectable="true" aria-label="{{ title|slugify }}s">
-    {% for choice in choices|slice:"1:" %}
-        <option{% if choice.selected %} selected{% endif %}
-            value="{{ choice.value }}"
-            data-key="{{ choice.key }}"
-            data-query="{{ choice.query_string|iriencode }}"
-            style="color:#000000"
-            {% if choice.selected %} aria-checked="true" {% else %} aria-checked="false" {% endif %}>
-            {{ choice.display }}
-        </option>
-    {% endfor %}
-    </select>
+    <div>
+        {% for choice in choices|slice:":1" %}
+            <a href="{{ choice.query_string|iriencode }}" title="Show {{ choice.display }}">Show {{ choice.display }}</a>
+        {% endfor %}
+        <select id="{{ title|slugify }}_select" multiple style="color:#000000;width:100%" aria-multiselectable="true" aria-label="{{ title|slugify }}s" size="{% if choices|length < 7 %} {{ choices|length }} {% else %} 8 {% endif %}">
+        {% for choice in choices|slice:"1:" %}
+            <option{% if choice.selected %} selected{% endif %}
+                value="{{ choice.value }}"
+                data-key="{{ choice.key }}"
+                data-query="{{ choice.query_string|iriencode }}"
+                style="color:#000000"
+                {% if choice.selected %} aria-checked="true" {% else %} aria-checked="false" {% endif %}>
+                {{ choice.display }}
+            </option>
+        {% endfor %}
+        </select>
+        <a id="{{ title|slugify }}_submit" href="" title="Filter by {{ title }}">Filter by {{ title }}</a>
+    </div>
 </div>
 
 <script>
-    // This function must have this name as it is referenced inside of https://github.com/raft-tech/django-admin-508/blob/main/admin_interface/static/admin_interface/508/dropdown-filter.js
-    // to allow custom filters to hook into the onClick event for the DAC `Apply Filters` button.
-    function custom_filter_callback($) {
-        // fetch data from selected options
-        var params = {};
-        var query = null;
-        $('select#{{ title|slugify }}_select > option:selected').each(function() {
-            var key = $(this).attr('data-key');
-            if (! params.hasOwnProperty(key)) params[key] = [];
-            params[key].push($(this).val());
-            if (!query) query = $(this).attr('data-query');
-        });
+    (function($) {
```

Comment:
why did you decide to create a template vs doing this as part of django-508?

Reply:
To support multiple multi-select filters on the same page in the admin we had to revert back to letting the mutliselect filters update the query string with their own "filter" buttons because there wasnt a way to use the existing callback method. The work that is coming out of [this](https://github.com/raft-tech/TANF-app/issues/3110) is going to provide a more full fledged solution to all our 508 filtering problems. The work here and the PR for 508 is just a work around until the linked issue is implemented.